src/config/project_config.py

```diff
@@ -113,7 +113,10 @@ def parse_arguments(self):
         parser.add_argument(
             "-d", "--debug", action="store_true", help="Enable debug output"
         )
-        return parser.parse_args()
+        # Only parse known arguments to avoid conflicts with other CLI arguments
+        args, _ = parser.parse_known_args()
```

Comment:
Concern about pipes or redirection?

Reply:
@nkanderson When I was trying to run my unittest I ran into an issue with the parser. The command was `PYTHONPATH=src/ python -m unittest tests/cache/test_cache_set.py -v`. Our argument parser does not have flags for `-m` or `-v`, so it didn't know what to do with these. Adding the` parse_known_args()` prevents any of the standard python interpreter arguments from causing conflicts with our custom ones we've defined for this project.

Reply:
Ahh, gotcha. So for a lot of our tests, mocking the logger and args is probably what we want to do, like in this [`setUp` and `tearDown` from the event_handler](https://github.com/nkanderson/ECE585-llc/blob/main/tests/utils/test_event_handler.py#L9-L32). Relevant lines below:

```python
def setUp(self):
    # Mocking the config logger and args for testing
    self.mock_logger = MagicMock()
    self.mock_args = MagicMock(silent=False, debug=True)

    # Patch the config logger and args
    patch(
        "config.project_config.config.get_logger", return_value=self.mock_logger
    ).start()
    patch(
        "config.project_config.config.get_args", return_value=self.mock_args
    ).start()

def tearDown(self):
    # Stop any patches done during the test
    patch.stopall()
```

Mocking the logger means we can test whether it was called without manually verifying the test output, for example:
```python
# Verify logger called with correct message for opcode 1
self.mock_logger.log.assert_called_once_with(
    LogLevel.DEBUG, "write request from L1 data cache"
)
```

Until we're getting into functional tests for the program as a whole, it's better to prevent any attempts by the sections of the program we're running from parsing args. Then in testing we could mock `cache_config` as well, or parameterize the `init` for the `Cache` class and just pass a `cache_config` instance directly in the test, which would provide flexibility in the test to try different values, including invalid config options.

fwiw, I think it may also be possible to write that command as `PYTHONPATH=src/ python -m unittest -v tests/cache/test_cache_set.py`.

Generally though, I think this change seems fine, but mocking parts of the project config in our tests is likely the approach we want in order to have control over test conditions and ability to test side effects, like with the logger.

Reply:
Created #48 for us to address this in a separate issue.